website/docs/opengl/c1/3-hello-texture.md

```diff
@@ -0,0 +1,388 @@
+Now for the fun part: rendering an image!
+
+To start this section, first download
+[silk.png](https://github.com/dotnet/Silk.NET/blob/477154b99564aac9cf82426754857103fb88d2ea/examples/CSharp/OpenGL%20Tutorials/Tutorial%201.3%20-%20Textures/silk.png).
+We will use this image in this tutorial!
+
+In most cases with OpenGL, textures are uploaded as a sequence of bytes, this is usually done with a layout of 4 bytes per pixel (Red, Green, Blue, and Alpha).
+The hard part is: we can't just upload the bytes of a .png or .jpg file! These formats have a lot of unnecessary data, like headers and most importantly
+compression, which the GPU generally cannot understand.
+
+To load an image file as a byte array, we will first need a external library. In our case, we will use StbImageSharp.
+
+First of all, install the library using NuGet. To do so, run the following command:
+```
+dotnet add package StbImageSharp
+```
+
+After installing, add the following reference on the top of your code file:
+```c#
+using StbImageSharp;
+```
+
+At the top of your class, declare a `uint` to store the ID of the OpenGL texture object:
+```c#
+private static uint _texture;
+```
+
+Returning to your `OnLoad` method, add this line at the end of the method to create, and bind the texture.
+That `ActiveTexture` call is telling OpenGL that we are wanting to use the first texture slot.
+```c#
+_texture = _gl.GenTexture();
+_gl.ActiveTexture(TextureUnit.Texture0);
+_gl.BindTexture(TextureTarget.Texture2D, _texture);
+```
+
+After that, we need to load the image. You can do it with the following line:
+```c#
+// ImageResult.FromMemory reads the bytes of the .png file and returns all its information!
+ImageResult result = ImageResult.FromMemory(File.ReadAllBytes("silk.png"), ColorComponents.RedGreenBlueAlpha);
+```
+
+Now, with the image in memory, we need to upload the data to the GPU texture. To dot hat, we need a pointer for our bytes, the width, and the height of
+the texture. None of it is hardcoded, you get all this information inside `ImageResult`.
+```c#
+// Define a pointer to the image data
+fixed (byte* ptr = result.Data)
+    // Here we use "result.Width" and "result.Height" to tell OpenGL about how big our texture is.
+    _gl.TexImage2D(TextureTarget.Texture2D, 0, InternalFormat.Rgba, (uint)result.Width,
+        (uint)result.Height, 0, PixelFormat.Rgba, PixelType.UnsignedByte, ptr);
+```
+
+You don't need to understand what all these values mean! The OpenGL documentation explains it all in depth. Right now
+let's just continue with the basics!
+
+Now, with the texture data sent off to the GPU, we need to configure the texture parameters. This is one of the most important steps. It is with these
+parameters that the shader knows how to sample colors from the texture using our provided texture coordinates.
+
+For now, you don't need to understand what these do. We will elaborate on it in the future. Just add these lines after you upload the texture data.
```

Comment:
Again I disagree.

Reply:
```suggestion
These will be explained in depth in [the following section on texture parameters](/#texture-parameters). Briefly, these affect how pixel data is read from the texture by the shader's `texture` function.
```

Note: I'm not sure if the link to /#texture-parameters will work like this

Reply:
i really want to use links like these here but idk how this markdown - to - html works so yeah, i think this is  good question to see in the future

Reply:
/# would send you to dotnet.github.io which would redirect you to dot.net. Remove the leading slash and it should be fine.

Reply:
o noticed and fixed it, its fine now